### PR TITLE
Resolve QueryException errors

### PR DESCRIPTION
**Pull request by profd2004**
Resolve QueryException error when service providers boot methods and other pre-migration operations uses translations.

### CODE REVIEW

File: src/TranslationLoaderManager.php [hunk 1]
```diff
@@ -20,7 +23,15 @@ public function load($locale, $group, $namespace = null): array
...
-        if (! is_null($namespace) && $namespace !== '*') {
+        $model = config('translation-loader.model');
+        $model = (new $model);
+        try {
+            DB::connection()->getPdo();
+        } catch (\Exception $e) {
+            return $fileTranslations;
+        }
+
+        if (! is_null($namespace) && $namespace !== '*' || ($model instanceof Model && !Schema::hasTable($model->getTable()))) {
```
Comment (freekmurze) [inline src/TranslationLoaderManager.php:34]: This condition is too complex. Could you extract this to a function that describes what is being checked? If possible, use early returns in that new function.

**Review by freekmurze — CHANGES_REQUESTED** [review]
Could you also add a test that proves this new logic works as intended?

File: src/TranslationLoaderManager.php [hunk 2]
```diff
@@ -43,4 +50,28 @@ protected function getTranslationsForTranslationLoaders(
...
+    protected function hasValidDbRequirements() {
+        $model = config('translation-loader.model');
+        $model = new $model;
+        if(!$model instanceof Model){
+            return false;
+        }
+
+        try {
+            DB::connection()->getPdo();
+        } catch (\Exception $e) {
```
Comment (freekmurze) [inline src/TranslationLoaderManager.php:67]: Could you import all namespace in your PR?

Comment (profd2004) [reply src/TranslationLoaderManager.php]: Did it!

**Comment by profd2004** [comment]
@freekmurze would you like me to do anything else on this?

**Comment by freekmurze** [comment]
Can you give me a concrete example why these changes are needed? No package users are complaining about any wrong behaviour. In which situation would you now get an error?

**Comment by drobee** [comment]
@freekmurze I've just installed this package on a Laravel 8 project and got the QueryException exception when composer tried to run the `php artisan package:discover --ansi` post autoload dump script.
Same happens when I try to publish the migrations of the package.

**Comment by freekmurze** [comment]
@drobee can you confirm that the changes in this PR solve that problem?

**Comment by drobee** [comment]
@freekmurze installing from `profd2004/laravel-translation-loader` `master` (325f73cbb2796768a283c2fb3f3e85ab80a8c1bf) solves this for me. However I haven't inspected the modifications in this PR so I don't know if it's the perfect solution.

File: .phpunit.result.cache [hunk 3]
```diff
@@ -0,0 +1 @@
...ionLoaders\DbTest::it_will_cache_all_translations";d:0.084;s:113:"Spatie\TranslationLoader\Test\TranslationLoaders\DbTest::it_flushes_the_cache_when_a_translation_has_been_created";d:0.086;s:113:"Spatie\TranslationLoader\Test\TranslationLoaders\DbTest::it_flushes_the_cache_when_a_translation_has_been_updated";d:0.084;s:113:"Spatie\TranslationLoader\Test\TranslationLoaders\DbTest::it_flushes_the_cache_when_a_translation_has_been_deleted";d:0.085;s:88:"Spatie\TranslationLoader\Test\TranslationLoaders\DbTest::it_can_work_with_a_custom_model";d:0.084;s:139:"Spatie\TranslationLoader\Test\TranslationLoaders\DbTest::it_will_throw_an_exception_if_the_configured_model_does_not_extend_the_default_one";d:0.087;s:106:"Spatie\TranslationLoader\Test\TranslationLoaders\DbTest::it_knows_when_language_lines_table_does_not_exist";d:0.115;s:93:"Spatie\TranslationLoader\Test\TranslationLoaders\DbTest::it_knows_when_no_db_connection_exist";d:0.124;s:125:"Spatie\TranslationLoader\Test\TranslationManagerTest::it_will_not_use_database_translations_if_the_provider_is_not_configured";d:0.108;s:98:"Spatie\TranslationLoader\Test\TranslationManagerTest::it_will_merge_translation_from_all_providers";d:0.101;}}}
```
Comment (freekmurze) [inline .phpunit.result.cache:1]: This file shouldn't be in this PR

File: src/TranslationLoaderManager.php [hunk 4]
```diff
@@ -43,4 +51,28 @@ protected function getTranslationsForTranslationLoaders(
             })
             ->toArray();
     }
+
+    /**
```
Comment (freekmurze) [inline src/TranslationLoaderManager.php:55]: Remove this comment, the function name is clear enough.

File: src/TranslationLoaderManager.php [hunk 5]
```diff
@@ -43,4 +51,28 @@ protected function getTranslationsForTranslationLoaders(
...
+     * Confirms that migration has been run and 
+     * current context has a valid DB connection.
+     */
+    protected function hasValidDbRequirements() {
+        $model = config('translation-loader.model');
+        $model = new $model;
+        if(!$model instanceof Model){
```
Comment (freekmurze) [inline src/TranslationLoaderManager.php:62]: Add a newline here to make the code breathe a little more.

File: tests/TestCase.php [hunk 6]
```diff
@@ -61,4 +61,19 @@ protected function createLanguageLine(string $group, string $key, array $text):
     {
         return LanguageLine::create(compact('group', 'key', 'text'));
     }
+
+    protected function callMethod($object, string $method , array $parameters = [])
+    {
+        try {
+            $className = get_class($object);
+            $reflection = new \ReflectionClass($className);
+        } catch (\ReflectionException $e) {
```
Comment (freekmurze) [inline tests/TestCase.php:70]: Import these namespaces

**Review by freekmurze — CHANGES_REQUESTED** [review]
Thanks for your work on this.

**Comment by faculezcano** [comment]
@profd2004

**Pull request closed without merging**